core/src/main/java/org/apache/calcite/sql/dialect/MysqlSqlDialect.java

```diff
@@ -240,6 +281,17 @@ public MysqlSqlDialect(Context context) {
       unparseListAggCall(writer, call, null, leftPrec, rightPrec);
       break;
 
+    case FORMAT_DATE:
+    case FORMAT_TIME:
+    case FORMAT_TIMESTAMP:
+    case FORMAT_DATETIME:
+      writer.print("DATE_FORMAT(");
```

Comment:
This feels a bit weird for me but I might miss some context.
Should this live in the definition of FORMAT_DATETIME that DATE_FORMAT is actually an aliases with the operands in different order + different formats elements?

Reply:
Like all of them should alias to the same canonical form so you can read the canonical form in every language idk if that makes sense

Reply:
Yeah that's open for discussion -- how to best swap a `FORMAT_TIMESTAMP` Google SQL style function call to any other dialect. Snowflake for example uses `TO_CHAR` or `TO_VARCHAR`. 

Maybe a generic entry in `SqlKind` called `GOOGLE_SQL_DATETIME_FORMAT`?
Could even be more generic like `DATETIME_FORMAT` and then each dialect would know how to unparse.

Reply:
Or better yet convert these BQ format functions into the standard `CAST(... as FORMAT <str>)` and unparse the `CAST` to `DATE_FORMAT`, `TO_CHAR` accordingly.